### PR TITLE
Make all pipelines triggerable via '/azp run name'.

### PR DESCRIPTION
**Pull request by sandreenko**
Extend https://github.com/dotnet/runtime/pull/42933.

The old `/azp run` issue is not actual anymore because it shows:
```
You have several pipelines (over 10) configured to build pull requests in this repository. Specify which pipelines you would like to run by using /azp run [pipelines] command. You can specify multiple pipelines using a comma separated list.
```

### CODE REVIEW

**Comment by sandreenko** [comment]
Thanks, @briansull  for pointing this out

**Review by trylek — APPROVED** [review]
Awesome, thank you!

**Comment by sandreenko** [comment]
actually, there are a bunch of jobs that are triggered for this PR that should not have been, let me see why.

**Comment by sandreenko** [comment]
ok, that because they have "trigger", for example, https://github.com/dotnet/runtime/blob/56ecd32333be78378626dc2effab46df5aa5f954/eng/pipelines/coreclr/jitrollingbuild.yml#L2-L9

@BruceForstall do you know how to change the trigger to work only for merged RPs? I thought `branches: include: - master` would do that but it does not.

**Comment by trylek** [comment]
Oh, interesting, I guess that may have been the reason why I chickened out of enabling some of the pipelines ;-).

**Comment by sandreenko** [comment]
The problem was in UI:
image

**Comment by trylek** [comment]
Well, I think there are two things: one is "Pull request enabled", that may indeed mean that the pipeline is automatically triggered upon pull requests (and it was probably just somehow force-blocked by the "pr:none" statement) and the other is "Build pull request from forks". I was under the impression that the latter is necessary so that individual devs can explicitly trigger these pipelines to run private tests on private branches in their forks of the repo, typically by using the <code>refs/pull/45931/head</code> pseudo-commit name; can you trigger a private run of the "runtime-coreclr jitrollingbuild" pipeline with Commit="refs/pull/45931" after unchecking "Build pull requests from forks of this repository"?

**Comment by sandreenko** [comment]
> Well, I think there are two things: one is "Pull request enabled", that may indeed mean that the pipeline is automatically triggered upon pull requests (and it was probably just somehow force-blocked by the "pr:none" statement) and the other is "Build pull request from forks". I was under the impression that the latter is necessary so that individual devs can explicitly trigger these pipelines to run private tests on private branches in their forks of the repo, typically by using the refs/pull/45931/head pseudo-commit name; can you trigger a private run of the "runtime-coreclr jitrollingbuild" pipeline with Commit="refs/pull/45931/head" after unchecking "Build pull requests from forks of this repository"?

I can https://dev.azure.com/dnceng/internal/_build/results?buildId=916971&view=results,
my current understanding is that a clean way of doing this would be to write a trigger in yml, like:
```
pr: only with a manual requires from a person with write access.
```
but I do not see such syntaxis in yml docs (https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=example%2Cparameter-schema), so what we want:
1. each pipeline can be triggered by `/azp run name` from a person with write access -> means we can't have `pr: none`;
2. we don't run pipelines for each PR -> because we can't setup it in yml we have to use UI for it.

**Comment by sandreenko** [comment]
Ok, I had to open a[nother PR ](https://github.com/dotnet/runtime/pull/45936)to check that I have not missed any jobs (and found that I missed libraries-gcstress0x3-gcstress0xc ), so what was tested:
1. PR from a fork does not trigger any extra pipelines;
2. PR from dotnet (not a fork) does not trigger any extra pipelines;
3. all pipelines can be run with `/azp run name`;
4. all pipeline still can be run manually via UI and `refs/pull/PR_number/head`

**Comment by trylek** [comment]
Sounds great, thanks for your diligence in testing all the possible scenarios!

**Comment by sandreenko** [comment]
@dotnet/runtime-infrastructure this PR changed a bunch of pipelines please ping me if you see an unexpected behavior after this.

File: eng/pipelines/runtime-official.yml
Comment (safern) [inline eng/pipelines/runtime-official.yml:23]: This should indeed be `pr: none` this should never run on PRs. This is the official build yml.

Comment (sandreenko) [reply eng/pipelines/runtime-official.yml:23]: it won't run unless somebody with write access manually triggers it, do we want to forbid it?  was /is it possible to trigger this pipeline using AzDo interface?

Comment (safern) [reply eng/pipelines/runtime-official.yml:23]: Se want to forbid it since it would fail if you run it from the public domain. There is not even a public pipeline pointing to it, so I would doubt azure devops would say there is no pipeline with that name. Anyway I think it is not a big of a deal since we can't run it anyway.

Comment (hoyosjs) [reply eng/pipelines/runtime-official.yml:23]: There's no pipeline using the definition in public, so you can't trigger this on a PR. It's not wrong per se, but it's safer to not let it run in case someone accidentally tried to. (Not sure it would even run correctly as it will try to access a bunch of secrets and tasks only available internally)

**Comment by BruceForstall** [comment]
The "jitrollingbuild" pipeline should never be triggerable from a PR.

**Pull request merged**
